### DISCUSSION TITLE
make logging to stderr available with vlib/log

### DISCUSSION THREAD

**Discussion by gergelyk** [vlang/v#18643] [Ideas]
### Describe the feature

By convention application logs typically should go to stderr. Make it possible with vlib/log.

### Use Case

I would like users to easily distinguish between computation results from my app (which go to stdout) and the logs.

### Proposed Solution

I would recommend to:
- implement `set_output_stream` method that allows to select between stdout and stderr.
- make stderr default (if such backward incompatibility is acceptable).

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.2 865c0e

### Environment details (OS name and version, etc.)

OS: macos, macOS, 12.6.1, 21G217
Processor: 10 cpus, 64bit, little endian, Apple M1 Pro
CC version: Apple clang version 14.0.0 (clang-1400.0.29.202)

**Comment by JalonSolov** [comment]
"typically"... where?  :-)  I've seen error logs go to stderr, but regular log messages often go to stdout.

It would, however, be nice to have the choice.

**Comment by gergelyk** [comment]
For instance 
```
curl -v example.org
```
writes html to stdout and logs to stderr.

Topic also discussed here: https://stackoverflow.com/a/4919110

**Comment by gergelyk** [comment]
But I agree, example above is not enough to say "typically". Anyways, having a choice would be nice.

**Comment by gergelyk** [comment]
Logging module in Python also seems to select stderr by default:

https://github.com/python/cpython/blob/3.11/Lib/logging/__init__.py#L1084

If only errors should go to stderr, selection of the stream would need to be dependent of the logging level. Personally, I think that this would overcomplicate though.

**Comment by justinwyer** [comment]
The current `log` implementation handles writing to stdout via println, would there be support for dropping that special use case in favour of using stdout/stderr as just another file within `log`?
